src/Symfony/Component/Cache/Adapter/OpCacheAdapter.php

```diff
@@ -0,0 +1,358 @@
+     * @var AdapterInterface
+     */
+    private $fallbackPool;
+
+    /**
+     * @param string $file The PHP file were values are cached.
+     *
+     * @return CacheItemPoolInterface
+     */
+    public static function createOnPhp7($file, CacheItemPoolInterface $fallbackPool)
+    {
+        // OPcache shared memory for arrays works only with PHP 7.0+
+        if (PHP_VERSION_ID >= 70000) {
+            if (!$fallbackPool instanceof AdapterInterface) {
+                $fallbackPool = new ProxyAdapter($fallbackPool);
+            }
+
+            return new static($file, $fallbackPool);
+        }
+
+        return $fallbackPool;
+    }
+
+    /**
+     * @param string           $file         The PHP file were values are cached.
+     * @param AdapterInterface $fallbackPool A pool to fallback on when an item is not hit.
+     * @param int              $umask
+     *
+     * @throws \InvalidArgumentException When the umask is invalid.
+     */
+    public function __construct($file, AdapterInterface $fallbackPool, $umask = 0002)
+    {
+        if (!is_int($umask)) {
+            throw new \InvalidArgumentException(sprintf('The parameter umask must be an integer, was: %s', gettype($umask)));
+        }
+
+        $this->file = $file;
+        $this->umask = $umask;
+        $this->fallbackPool = $fallbackPool;
+        $this->createCacheItem = \Closure::bind(
+            function ($key, $value, $isHit) {
+                $item = new CacheItem();
+                $item->key = $key;
+                $item->value = $value;
+                $item->isHit = $isHit;
+
+                return $item;
+            },
+            $this,
+            CacheItem::class
+        );
+    }
+
+    /**
+     * Store a static array of cached values.
+     *
+     * @param array $values The cached values.
+     */
+    public function warmUp(array $values)
+    {
+        if (file_exists($this->file)) {
+            if (!is_file($this->file)) {
+                throw new InvalidArgumentException(sprintf('Cache path exists and is not a file: %s', $this->file));
+            }
+
+            if (!is_writable($this->file)) {
+                throw new InvalidArgumentException(sprintf('Cache file is not writable: %s', $this->file));
+            }
+        } else {
+            $directory = dirname($this->file);
+
+            if (!is_dir($directory) && !@mkdir($directory, 0777 & (~$this->umask), true)) {
+                throw new InvalidArgumentException(sprintf('Cache directory does not exist and cannot be created: %s', $directory));
+            }
+
+            if (!is_writable($directory)) {
+                throw new InvalidArgumentException(sprintf('Cache directory is not writable: %s', $directory));
+            }
+        }
+
+        $dump = <<<'EOF'
+<?php
+
+// This file has been auto-generated by the Symfony Cache Component.
+
+return array(
+
+
+EOF;
+
+        foreach ($values as $key => $value) {
+            CacheItem::validateKey($key);
```

Comment:
because PHP casts numeric string keys to ints, we should cast ints to string here:
`is_int($key) ? (string) $key : $key);